presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java

```diff
@@ -1635,6 +1635,33 @@ public void testColumnFilter()
         assertQuery(session, format("SELECT c_boolean, c_bigint, c_double, c_varchar, c_varbinary FROM %s WHERE c_varbinary = to_ieee754_64(1)", tmpTableName));
     }
 
+    @Test
+    public void testUuid()
+    {
+        assertQuerySucceeds("SELECT uuid()");
+        assertQuerySucceeds("SELECT ARRAY[uuid(), uuid(), uuid()]");
+        assertQuerySucceeds("SELECT ARRAY[uuid(), uuid(), uuid()][1] AS first_col");
+
+        // Typeof uuid.
+        assertQuery("SELECT typeof(uuid())");
+        assertQuery("SELECT typeof(array_constructor(uuid(), uuid()))");
+
+        // Valid UUID.
+        assertQuery("SELECT cast('33355449-2c7d-43d7-967a-f53cd23215ad' AS uuid)");
```

Comment:
Do these queries run on coordinator or worker?

Reply:
@mbasmanova : 

Am following up with tests for CTAS + INSERTs etc. These SQL are largely co-ordinator.
The review went out too soon.

Reply:
@mbasmanova : UUID would need a few more changes to have a better user experience. It has equality comparison, but lt, lte, gt, gte, between comparisons are not available. So authoring SQL has limitations.

Reply:
@aditi-pandit Aditi, thank you for sharing these findings. Will you be adding missing functionality to Velox?

Reply:
@mbasmanova : Created issues https://github.com/facebookincubator/velox/issues/10584 and https://github.com/facebookincubator/velox/issues/10585. We will follow up there.